util/device_sw_utils/extract_sw_logs.py

```diff
@@ -177,10 +179,8 @@ def extract_sw_logs(elf_file, logs_fields_section, ro_sections):
                 size = int(section.header['sh_size'])
                 data = section.data()
                 ro_contents.append((base_addr, size, data))
-            else:
-                print("Error: {} section not found in {}".format(
-                    ro_section, elf_file))
-                sys.exit(1)
+            # If there is no ro section then we will end up generating empty
+            # database files which is OK, and keeps the SW build from failing.
```

Comment:
huh...so we didn't actually need the `ro section`? it was just a check?

Reply:
yeah from a SW side you don't need it there is nothing to put in it, as @alphan described in #14636 when he implemented the `rom_e2e_shutdown_exception_c test` in #14627. On the [SV side](https://cs.opensource.google/opentitan/opentitan/+/master:hw/dv/sv/sw_logger_if/sw_logger_if.sv) it looks like [it won't do anything](https://cs.opensource.google/opentitan/opentitan/+/master:hw/dv/sv/sw_logger_if/sw_logger_if.sv;drc=8169fb4bc0331a177fab0b737539d3e274c721f8;l=352) if there is an empty database file. So just I figured just let it generate empty files.

Reply:
is this [this part](https://cs.opensource.google/opentitan/opentitan/+/master:hw/dv/sv/sw_logger_if/sw_logger_if.sv;l=193)? Just ..doesn't do anything?

Reply:
The `rodata` section parsing was added to this script to support the `%s` modifier. For example, `LOG_INFO("chip_info = %s", chip_info)`. 

The way it works is - the dv logger (SW) writes the address of the string when writing the arg values, and the SV backend then find the value of the string at that address by looking up the rodata section. If the string is expected to be in some other section, then that must be passed to this script via the `--rodata-sections` switch. 

Without properly supporting this, DV users will have a degraded experience (`%s` will not be properly supported).

Reply:
Proper support would mean updating the bazel rules to pass all appropriate sections using the `--rodata-sections` switch for the script to prepare a database read by the SV side.

Reply:
This does not remove the rodata section parsing. This just make it so if the program has no rodata section (as in there are no strings stored to print, as is the case with [this](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/e2e/rom_e2e_shutdown_exception_c_test.c?q=shutdown_exception_c) ROM E2E test program) the script wont error out. Rather it will create empty database files to feed to the SV side so it will have something to "load", and since there are no logs anywhere in the program it will never log anything.

Agreed we should also update the [bazel rule](https://cs.opensource.google/opentitan/opentitan/+/master:rules/opentitan.bzl;drc=c8a7ff764fa53dcdfd193789932f50b3375b897d;l=493) to pass more `--rodata-sections`. I can do that in a follow up PR.

Reply:
Actually the more I think about this, it does not make sense to update the bazel rule to pass more sections to the `--rodata-sections` flag because the `LOG_INFO` macro creates a `const char *format` string [which will always reside in the `.rodata` section](https://gist.github.com/jesstess/1861075#file-elf_strings-L32), and since this macro is the only supported mechanism to backdoor log messages in DV this is the only section we will ever need to look for said strings, no?

Reply:
See my first comment about supporting the `%s` format specifier. It's not just the `format` string - its the string args as well, which can reside in any section. The test ROM logs the `chip_info` for instance, which resides in a separate `.chip_info` section and not in `.rodata`. 
 
I have properly fixed this by eliminating the `--ro-sections` altogether - the script now looks up all available sections in the ELF rather than being told which sections to look into. PR coming soon.

Reply:
> This does not remove the rodata section parsing. This just make it so if the program has no rodata section (as in there are no strings stored to print, as is the case with [this](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/e2e/rom_e2e_shutdown_exception_c_test.c?q=shutdown_exception_c) ROM E2E test program) the script wont error out. Rather it will create empty database files to feed to the SV side so it will have something to "load", and since there are no logs anywhere in the program it will never log anything.
> 

Right - my clarification was for TimC who I thought may have gotten the wrong idea that the script didn't do anything with the `rodata` section at all, which is not true. I was already made aware of the issue in #14636 and had a fix in mind.

Reply:
> > This does not remove the rodata section parsing. This just make it so if the program has no rodata section (as in there are no strings stored to print, as is the case with [this](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/e2e/rom_e2e_shutdown_exception_c_test.c?q=shutdown_exception_c) ROM E2E test program) the script wont error out. Rather it will create empty database files to feed to the SV side so it will have something to "load", and since there are no logs anywhere in the program it will never log anything.
> 
> Right - my clarification was for TimC who I thought may have gotten the wrong idea that the script didn't do anything with the `rodata` section at all, which is not true. I was already made aware of the issue in #14636.

Reply:
ah right the args, my apologies!